Formula/crystal.rb

```diff
@@ -75,17 +80,27 @@ def install
 
     (buildpath/"boot").install resource("boot")
     ENV.append_path "PATH", "boot/bin"
-    ENV.append_path "CRYSTAL_LIBRARY_PATH", Formula["bdw-gc"].opt_lib
-    ENV.append_path "CRYSTAL_LIBRARY_PATH", ENV["HOMEBREW_LIBRARY_PATHS"]
-    ENV.append_path "CRYSTAL_LIBRARY_PATH", Formula["libevent"].opt_lib
+    ENV["CRYSTAL_LIBRARY_PATH"] = ENV["HOMEBREW_LIBRARY_PATHS"]
+    ENV.append_path "CRYSTAL_LIBRARY_PATH", MacOS.sdk_path_if_needed/"usr/lib" if MacOS.sdk_path_if_needed
     ENV.append_path "LLVM_CONFIG", llvm.opt_bin/"llvm-config"
 
+    crystal_install_dir = libexec
+    # Avoid embedding HOMEBREW_PREFIX references in `crystal` binary.
+    config_library_paths = ENV["CRYSTAL_LIBRARY_PATH"].gsub(
+      HOMEBREW_PREFIX,
+      "\\$$ORIGIN/#{HOMEBREW_PREFIX.relative_path_from(crystal_install_dir)}",
+    )
+    crystal_build_opts = [
+      "release=true",
+      "FLAGS=--no-debug",
+      "CRYSTAL_CONFIG_LIBRARY_PATH=#{config_library_paths}",
```

Comment:
This should be the (relatively simple) configuration that configures both paths relative to the install location of the binary and should make it completely relocatable without any overrides in the wrapper script:
```suggestion
      "CRYSTAL_CONFIG_LIBRARY_PATH=\\$$ORIGIN/#{HOMEBREW_PREFIX.relative_path_from(crystal_install_dir)}/lib",
      "CRYSTAL_CONFIG_PATH=\\$$ORIGIN/#{pkgshare.relative_path_from(crystal_install_dir)}/src",
```